unstructured_ingest/v2/processes/connectors/outlook.py

```diff
@@ -0,0 +1,239 @@
+import hashlib
+import time
+from dataclasses import dataclass, field
+from datetime import timezone
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Coroutine, Generator
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.logger import logger
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    FileData,
+    Indexer,
+    IndexerConfig,
+    download_responses,
+)
+from unstructured_ingest.v2.interfaces.file_data import FileDataSourceMetadata, SourceIdentifiers
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+MAX_EMAILS_PER_FOLDER = 1_000_000  # Maximum number of emails per folder
+
+if TYPE_CHECKING:
+    from office365.graph_client import GraphClient
+    from office365.outlook.mail.folders.folder import MailFolder
+    from office365.outlook.mail.messages.message import Message
+
+
+CONNECTOR_TYPE = "outlook"
+
+
+class OutlookAccessConfig(AccessConfig):
+    client_credential: str = Field(description="Azure AD App client secret", alias="client_cred")
+
+
+class OutlookConnectionConfig(ConnectionConfig):
+    access_config: Secret[OutlookAccessConfig]
+    client_id: str = Field(description="Azure AD App client ID")
+    tenant: str = Field(
+        default="common", description="ID or domain name associated with your Azure AD instance"
+    )
+    authority_url: str = Field(
+        default="https://login.microsoftonline.com",
+        description="Authentication token provider for Microsoft apps",
+    )
+
+    @requires_dependencies(["msal"])
+    def _acquire_token(self):
+        """Acquire token via MSAL"""
+        from msal import ConfidentialClientApplication
+
+        # NOTE: It'd be nice to use `msal.authority.AuthorityBuilder` here paired with AZURE_PUBLIC
+        # constant as default in the future but they do not fit well with `authority_url` right now
+        authority_url = f"{self.authority_url.rstrip('/')}/{self.tenant}"
+        app = ConfidentialClientApplication(
+            authority=authority_url,
+            client_id=self.client_id,
+            client_credential=self.access_config.get_secret_value().client_credential,
+        )
+        token = app.acquire_token_for_client(scopes=["https://graph.microsoft.com/.default"])
+        return token
+
+    @requires_dependencies(["office365"], extras="graph")
```

Comment:
I think this should be:
`@requires_dependencies(["office365"], extras="outlook")`
which results in prompting the user to 
`pip install "unstructured-ingest[outlook]"`
if they need the dependencies.

If you change it to `graph` you would have to make a bunch of changes in the requirements files etc... because when I was prompted to `pip install "unstructured-ingest[graph]"` nothing happened.

Reply:
You're right, I've confused it when writing from V1 and was actually sure it's been that way there for some reason.
I've updated that here and in the second usage of this function for `msal` import.
I've also added a doc-string to the decorator.